### DISCUSSION TITLE
Improve Craft 4 upgrade docs

### DISCUSSION THREAD

**Discussion by eduardoalonsoalbella** [craftcms/docs#468] [Ideas]
Hi, I'm currently upgrading to Craft 4, my main issue is that a few things have been removed and I think an alternative should have been proposed in the documentation.

The docs say: https://craftcms.com/docs/4.x/extend/updating-plugins.html#control-panel-templates

```
Removed the assets/_edit control panel template.
Removed the categories/_edit control panel template.
Removed the entries/_edit control panel template.
Removed the _layouts/element control panel template.
```

(the same with template hooks)

In our code we are doing some operations in case one of those templates are loaded.

```
        Event::on(
            View::class,
            View::EVENT_BEFORE_RENDER_TEMPLATE,
            function(TemplateEvent $event) {
                if ($event->template != 'entries/_edit') {
                    return;
                }
                // do something
            }
        );
```

But no alternative is proposed, in future breaking compatibility changes, an alternative should always be proposed in the docs

A good example of good upgrade docs is: https://laravel.com/docs/9.x/upgrade

**Comment by brandonkelly** [comment]
What are you doing from that event, exactly?

**Comment by eduardoalonsoalbella** [comment]
Hi @brandonkelly 

Primarily, adding a few field in the side panel, using the hook "cp.entries.edit" and adding a few js/css files when editing entries.

For me the ideal would be, to have an example that shows the equivalent of how do thing the new way, implementing a similar behaviour of the hooks that where removed 

Another one would be, how to perform an action when an entry is being edited, now that the _edit templates are gone, we need to find a way to do that

```
        Event::on(
            View::class,
            View::EVENT_BEFORE_RENDER_TEMPLATE,
            function(TemplateEvent $event) {
                if ($event->template != 'entries/_edit') {
                    return;
                }
               // do something
              Craft::$app->view->hook('cp.entries.edit' ....
```
Do you have a link that describes how UI elements in Craft work? example of show a table (a simple one), set up an action end point, and configure the table to request the data to that endpoint


Thanks for asking
